crates/transpiler/src/passes/basis_translator/mod.rs

```diff
@@ -689,57 +680,104 @@ fn replace_node(
-                        let bound_param = param_obj.as_ref().clone().into_bound_py_any(py)?;
-
-                        let exp_params = param.iter_parameters()?;
-                        let bind_dict = PyDict::new(py);
-                        for key in exp_params {
-                            bind_dict.set_item(key.clone(), parameter_map.get_item(key)?)?;
-                        }
-                        let mut new_value: Bound<PyAny>;
-                        let comparison = bind_dict.values().iter().any(|param| {
-                            param
-                                .is_instance(PARAMETER_EXPRESSION.get_bound(py))
-                                .is_ok_and(|x| x)
-                        });
-                        if comparison {
-                            new_value = bound_param.clone();
-                            for items in bind_dict.items() {
-                                new_value = new_value.call_method1(
-                                    intern!(py, "assign"),
-                                    items.downcast::<PyTuple>()?,
-                                )?;
+                        let new_value: Param = {
+                            let mut bind_dict = HashMap::new();
+                            for key in param_obj.iter_symbols() {
+                                bind_dict.insert(key.clone(), parameter_map[key].clone());
                             }
-                        } else {
-                            new_value =
-                                bound_param.call_method1(intern!(py, "bind"), (&bind_dict,))?;
-                        }
-                        let eval = new_value.getattr(intern!(py, "parameters"))?;
-                        if eval.is_empty()? {
-                            new_value = new_value.call_method0(intern!(py, "numeric"))?;
-                        }
-                        new_params.push(new_value.extract()?);
+                            let mut new_value: Arc<ParameterExpression>;
+                            if bind_dict
+                                .values()
+                                .any(|param| matches!(param, Param::ParameterExpression(_)))
+                            {
+                                new_value = param_obj.clone();
+                                for (symbol, val) in bind_dict.iter() {
+                                    let Param::ParameterExpression(val) = val else {
+                                        continue;
+                                    };
+                                    let map: HashMap<Symbol, ParameterExpression> =
+                                        [(symbol.clone(), ParameterExpression::clone(val))]
+                                            .into_iter()
+                                            .collect();
+                                    new_value = new_value
+                                        .subs(&map, false)
```

Comment:
If we're sure that the `map` is valid, we could change this to
```suggestion
                                        .subs(&map, true)
```
which would skip some symbol checks and be faster (same in other places for `bind`)

Reply:
Added in 2be5b1c53f7727171a149982ab6e9ccda285e048